apps/mobile-wallet/src/screens/LandingScreen.tsx

```diff
@@ -57,11 +59,25 @@ const LandingScreen = ({ navigation, ...props }: LandingScreenProps) => {
   const insets = useSafeAreaInsets()
   const theme = useTheme()
   const { t } = useTranslation()
+  const isWalletUnlocked = useAppSelector((s) => s.wallet.isUnlocked)
 
   const { width, height } = Dimensions.get('window')
   const [dimensions, setDimensions] = useState({ width, height })
   const [showNewWalletButtons, setShowNewWalletButtons] = useState(false)
 
+  // Normally, when the app is unlocked, this screen is not in focus. However, under certain conditions we end up with
+  // an unlocked wallet and no screen in focus at all. This happens when:
+  // 1. the auto-lock is set to anything but "Fast"
+  // 2. the user manually kills the app before the auto-lock timer completes
+  // 3. the WalletConnect feature is activated
+  // Since there is no screen in focus and since the default screen set in the RootStackNavigation is this screen, we
+  // need to navigate back to the dashboard.
+  useFocusEffect(
+    useCallback(() => {
+      if (isWalletUnlocked) resetNavigation(navigation)
+    }, [isWalletUnlocked, navigation])
+  )
+
```

Comment:
I had the idea of setting the dashboard screen as the initial screen of the RootStackNavigation. I need to think more about this because it will have quite some implications for our auth logic. Fight for another day.

Reply:
Yeah we'll need to test that thoroughly.